client/components/UI/Form/SelectMetaTermsInput/SelectFieldPopup.tsx

```diff
@@ -280,10 +280,16 @@ export class SelectFieldPopup extends React.Component<IProps, IState> {
 
         const valueNoCase = val.toLowerCase();
         const searchKey = this.props.searchKey ?? 'name';
-        let searchResults = this.props.options.filter((opt) => (
-            opt[searchKey].toLowerCase().substr(0, val.length) === valueNoCase ||
-                opt[searchKey].toLowerCase().indexOf(valueNoCase) >= 0
-        ));
+
+        let searchResults = this.props.options.filter((opt) => {
```

Comment:
try removing code duplication here: use a conditional for setting a function that would get a label in a correct language, but reuse the same label comparison logic without duplicating code